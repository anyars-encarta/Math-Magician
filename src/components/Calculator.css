*,
body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Roboto',
    'Oxygen',
    'Ubuntu',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.main {
  float: right;
  width: 50%;
}

.header,
button {
  font-size: 20px;
}

.header {
  text-align: center;
  font-size: 30px;
}

.calculator {
  display: grid;
  grid-template-rows: minmax(4rem, auto) repeat(5, 1fr);
  grid-template-columns: repeat(4, 5rem);
  padding: 5px;
  align-items: center;
  justify-content: center;
}

.cal-answer {
  grid-column: 1 / span 4;
  background-color: rgba(179, 173, 173, 0.616);
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 5px;
  font-size: 2.5em;
  margin-right: 1px;
  width: 20rem;
  height: 3.5rem;
}

.answer {
  font-size: 2rem;
  color: rgb(255, 255, 255);
  font-family: 'Dancing Script', cursive;
  text-align: end;
}

.zero {
  grid-column: 1 / span 2;
  cursor: pointer !important;
}

button {
  height: 60px;
  font-weight: bold;
}

.side {
  background-color: chocolate;
  color: #fff;
}
