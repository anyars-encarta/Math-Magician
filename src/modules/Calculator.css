*,
body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Roboto',
    'Oxygen',
    'Ubuntu',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.main {
  width: 60%;
  margin: 0 auto;
}

.header,
button {
  font-size: 20px;
}

.header {
  text-align: center;
  font-size: 50px;
  color: #fff;
}

.calculator {
  display: grid;
  grid-template-rows: minmax(4rem, auto) repeat(5, 1fr);
  grid-template-columns: repeat(4, 5rem);
  padding: 5px;
  align-items: center;
  justify-content: center;
}

.calcSection {
  display: flex;
  justify-content: space-between;
}

.header,
.calcSection h2 {
  text-shadow: 1px 1px 2px red, 0 0 1em blue, 0 0 0.2em blue;
}

.calcSection h2 {
  color: white;
  font-size: 40px;
}

.cal-answer {
  grid-column: 1 / span 4;
  background-color: rgba(179, 173, 173, 0.616);
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 5px;
  font-size: 2.5em;
  margin-right: 1px;
  width: 20rem;
  height: 3.5rem;
  color: #ffff;
  text-shadow: 1px 1px 2px black;
}

.answer {
  font-size: 2rem;
  color: rgb(255, 255, 255);
  font-family: 'Dancing Script', cursive;
  text-align: end;
}

.zero {
  grid-column: 1 / span 2;
  cursor: pointer !important;
}

button {
  height: 60px;
  font-weight: bold;
}

.side {
  background-color: chocolate;
  color: #fff;
}
