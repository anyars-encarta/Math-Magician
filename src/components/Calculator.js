import React, { useState } from 'react';
import CalculatorButton from './CalculatorButton';
import calculate from '../logic/calculate';
import '../modules/Calculator.css';

function Calculator() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const newData = calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  };

  return (
    <div className="main">
      <h1 className="header">Calculator</h1>
      <div className="calcSection">
        <h2>Let&apos;s do some math!</h2>
        <div className="calculator">
          <div className="cal-answer" data-testid="calc-answer">
            {calculatorData.next || calculatorData.total || '0'}
          </div>
          <CalculatorButton onClick={() => handleClick('AC')}>AC</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('+/-')}>+/-</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('%')}>%</CalculatorButton>
          <CalculatorButton className="side" onClick={() => handleClick('รท')}>/</CalculatorButton>

          <CalculatorButton onClick={() => handleClick('7')}>7</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('8')}>8</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('9')}>9</CalculatorButton>
          <CalculatorButton className="side" onClick={() => handleClick('x')}>X</CalculatorButton>

          <CalculatorButton onClick={() => handleClick('4')}>4</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('5')}>5</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('6')}>6</CalculatorButton>
          <CalculatorButton className="side" onClick={() => handleClick('-')}>-</CalculatorButton>

          <CalculatorButton onClick={() => handleClick('1')}>1</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('2')}>2</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('3')}>3</CalculatorButton>
          <CalculatorButton className="side" onClick={() => handleClick('+')}>+</CalculatorButton>

          <CalculatorButton className="zero" onClick={() => handleClick('0')}>0</CalculatorButton>
          <CalculatorButton onClick={() => handleClick('.')}>.</CalculatorButton>
          <CalculatorButton className="side" onClick={() => handleClick('=')}>=</CalculatorButton>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
