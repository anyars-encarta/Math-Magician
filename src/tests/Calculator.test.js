import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

// Test the Calculator component using Jest snapshots
describe('Calculator', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// Test user interactions using React Testing Library
describe('Calculator interactions', () => {
  it('should handle button clicks correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    const answerDisplay = getByTestId('calc-answer');

    // Simulate button clicks
    fireEvent.click(getByText('7'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));

    // Check if the answer is displayed correctly
    expect(answerDisplay.textContent).toBe('10');
  });
});
