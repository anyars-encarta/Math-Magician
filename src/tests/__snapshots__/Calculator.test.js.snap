// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <h1
      class="header"
    >
      Calculator
    </h1>
    <div
      class="calcSection"
    >
      <h2>
        Let's do some math!
      </h2>
      <div
        class="calculator"
      >
        <div
          class="cal-answer"
          data-testid="calc-answer"
        >
          0
        </div>
        <button
          class=""
          type="button"
        >
          AC
        </button>
        <button
          class=""
          type="button"
        >
          +/-
        </button>
        <button
          class=""
          type="button"
        >
          %
        </button>
        <button
          class="side"
          type="button"
        >
          /
        </button>
        <button
          class=""
          type="button"
        >
          7
        </button>
        <button
          class=""
          type="button"
        >
          8
        </button>
        <button
          class=""
          type="button"
        >
          9
        </button>
        <button
          class="side"
          type="button"
        >
          X
        </button>
        <button
          class=""
          type="button"
        >
          4
        </button>
        <button
          class=""
          type="button"
        >
          5
        </button>
        <button
          class=""
          type="button"
        >
          6
        </button>
        <button
          class="side"
          type="button"
        >
          -
        </button>
        <button
          class=""
          type="button"
        >
          1
        </button>
        <button
          class=""
          type="button"
        >
          2
        </button>
        <button
          class=""
          type="button"
        >
          3
        </button>
        <button
          class="side"
          type="button"
        >
          +
        </button>
        <button
          class="zero"
          type="button"
        >
          0
        </button>
        <button
          class=""
          type="button"
        >
          .
        </button>
        <button
          class="side"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
